### ISSUE TITLE
Remove legacy cookies

### ISSUE THREAD

**Issue by jonkoops** [keycloak/keycloak#16770]
`_LEGACY` cookies was introduced in Keycloak as a work-around to SameSite enforcement in browsers. As time has past SameSite for cookies is widely supported and this work-around is no longer required.

One caveat to the above is that the `_LEGACY` cookies also served another purpose, which is they where available in an insecure context, while the regular cookies are not. This results in login not working as expected when accessing a Keycloak deployment not through `localhost`, but through `http` rather than `https`. This is of course not recommended at all in production, but frequently used during development or testing. An alternative to the above is conditionally set the `secure` flag on cookies only when access over a secure context, as well as changing `SameSite=None` cookies to `SameSite=Lax` as user-agents will not send cookies with `SameSite=none` if they do not have the `secure` flag set.

**Comment by Wimmerke** [comment]
Jon thx

but all cookies need to be configurable and should have the secure option on by default

So if legacy, make it possible to disable the cookie (this will also help to check for future changes, with this cookie off)

For the other cookies, please keep in mind that these must be always
- OR secured by default
- OR configurable

hope this is available in the next update. As this impact the security score of the services in scope of compliance or security scans.

**Comment by jonkoops** [comment]
Agreed, but this is not part of this issue. Please keep the discussion relevant to the issue at hand.

**Comment by mposolda** [comment]
@jonkoops I suppose we need to postpone to 24 considering that we don't even have https://github.com/keycloak/keycloak/issues/16181 fixed?

**Comment by jonkoops** [comment]
@mposolda Yeah, this will have to be postponed. If someone could focus some effort on #16181 and #16747 that would be great. I'd really like to remove this old cruft.

**Comment by jonkoops** [comment]
Since this is blocked on the test suite refactor we'll have to bump this down the line for now. I am moving it to the next milestone so we don't forget about it.

**Comment by Wimmerke** [comment]
> Since this is blocked on the test suite refactor we'll have to bump this down the line for now. I am moving it to the next milestone so we don't forget about it.

thanks Jonkoops to keep this on the radar. much appreciated.

**Comment by jonkoops** [comment]
Bumping this down to v27 as we'll have to have a deprecation phase in v26 first.
